src/memory/memory.cc

```diff
@@ -1,44 +1,44 @@
-#include <algorithm>
-
 #include "memory/memory.hh"
 
 namespace sim {
 
-Word Memory::loadWord(Addr addr) {
-  stats.numLoads++;
-  Word word = kDummyWord;
-
-  auto it = mem.find(addr);
-  if (it != mem.end())
-    word = it->second;
-  else
-    word = pageFaultHandle(addr);
-
-  return word;
-}
-
-void Memory::storeWord(Addr addr, Word word) {
-  stats.numStores++;
-  mem[addr] = word;
-}
-
-Word Memory::pageFaultHandle(Addr addr) {
-  (void)addr;
-  stats.numPageFaults++;
-  std::cout << "Placeholder for page fault" << std::endl;
-  return kDummyWord;
-}
-
-std::size_t Memory::getCurrMemSize() const { return mem.size(); }
-
 void Memory::printMemStats(std::ostream &ost) const {
   ost << "Memory stats:" << std::endl;
   ost << "Loads: " << stats.numLoads << std::endl;
   ost << "Stores: " << stats.numStores << std::endl;
-  ost << "PageFaults: " << stats.numPageFaults << std::endl;
   ost << "End of memory stats." << std::endl;
 }
 
 const Memory::MemoryStats &Memory::getMemStats() const { return stats; }
 
+PhysMemory::listIt PhysMemory::pageTableLookup(const AddrSections &sect,
+                                               MemoryOp op) {
+  using MemOp = PhysMemory::MemoryOp;
+  auto it_P1 = pageTable.find(sect.index_pt1);
+  if (it_P1 == pageTable.end()) {
+    if (op == MemOp::LOAD)
+      throw std::runtime_error("Load on unmapped region in physical mem (P1)");
+    else {
+      pageStorage.emplace_back(Page{});
```

Comment:
```suggestion
      pageStorage.emplace_back();
```
`emplace_back()` accepts arguments of underlying type ctor via parameter pack, but since it can be empty, you can create default argument by calling `emplace_back` without arguments

Reply:
Also, you can commit this change by clicking commit button right under my change suggestion